sw/airborne/subsystems/ins/ins_vectornav.h

```diff
@@ -0,0 +1,186 @@
+ */
+#ifndef INS_VECTORNAV_H
+#define INS_VECTORNAV_H
+
+// Subsystems
+#include "subsystems/imu.h"
+#include "subsystems/gps.h"
+#include "subsystems/ahrs.h"
+#include "subsystems/ins.h"
+
+// Peripheral
+#include "peripherals/vn200_serial.h"
+
+// Geodetic / Math
+#include "math/pprz_algebra.h"
+#include "math/pprz_geodetic_int.h"
+#include "math/pprz_algebra_float.h"
+#include "math/pprz_geodetic_int.h"
+#include "math/pprz_isa.h"
+
+// Generated
+#include "generated/airframe.h"
+#include "generated/flight_plan.h"
+
+// Systime
+#include "mcu_periph/sys_time.h"
+
+// Abi
+#include "subsystems/abi.h"
+
+#if GPS_USE_LATLONG
+/* currently needed to get nav_utm_zone0 */
+#include "subsystems/navigation/common_nav.h"
+#include "math/pprz_geodetic_float.h"
+#endif
+
+/*
+ * IMU Subsystem defines
+ */
+extern void imu_vectornav_event(void);
+
+#if !USE_NPS
+#define ImuEvent imu_vectornav_event
+#endif /* Only for non-NPS targets */
+
+/*
+ * IMU scalling defaults - still needed, although not used
+ * Filled in with dummy values (ones)
+ */
+#if !defined IMU_GYRO_P_SENS & !defined IMU_GYRO_Q_SENS & !defined IMU_GYRO_R_SENS
+#define IMU_GYRO_P_SENS 1
+#define IMU_GYRO_P_SENS_NUM 1
+#define IMU_GYRO_P_SENS_DEN 1
+#define IMU_GYRO_Q_SENS 1
+#define IMU_GYRO_Q_SENS_NUM 1
+#define IMU_GYRO_Q_SENS_DEN 1
+#define IMU_GYRO_R_SENS 1
+#define IMU_GYRO_R_SENS_NUM 1
+#define IMU_GYRO_R_SENS_DEN 1
+#endif
+
+#if !defined IMU_ACCEL_X_SENS & !defined IMU_ACCEL_Y_SENS & !defined IMU_ACCEL_Z_SENS
+#define IMU_ACCEL_X_SENS 1
+#define IMU_ACCEL_X_SENS_NUM 1
+#define IMU_ACCEL_X_SENS_DEN 1
+#define IMU_ACCEL_Y_SENS 1
+#define IMU_ACCEL_Y_SENS_NUM 1
+#define IMU_ACCEL_Y_SENS_DEN 1
+#define IMU_ACCEL_Z_SENS 1
+#define IMU_ACCEL_Z_SENS_NUM 1
+#define IMU_ACCEL_Z_SENS_DEN 1
+#endif
+
+
+/*
+ * GPS subsystem defines
+ */
+#if !USE_NPS
+#define GpsEvent(_x) ;
+#endif /* Only for non-NPS targets */
+
+/*
+ * AHRS subsystem
+ * - uses ins_vectonav_wrapper.h
+ */
+
+/*
+ * INS subsystem defines
+ */
+#ifndef DefaultInsImpl
+#define DefaultInsImpl ins_vectornav
+#endif
+
+
+
+
+// Ins implementation state (fixed point)
+struct InsVectornav {
+  struct LtpDef_i  ltp_def; // initial position
+  bool_t           ltp_initialized; // status indicator
+
+  // output LTP NED for telemetry messages
+  struct NedCoor_i ltp_pos;
+  struct NedCoor_i ltp_speed;
+  struct NedCoor_i ltp_accel;
+
+  struct LlaCoor_f lla_pos;
+
+  struct NedCoor_f ltp_accel_f;
+
+  // baro [height above ground]
+  float baro_z;  ///< z-position calculated from baro in meters (z-down)
+  float qfe;
+
+  // Packet data
+  struct VNPacket vn_packet;///< Packet struct
+  enum VNStatus vn_status;  ///< VN status
+  float vn_freq;            ///< data frequency
+  uint16_t vn_chksm;         ///< aux variable for checksum
+  uint32_t vn_time;          ///< VN time stamp
+  uint32_t vn_ltime;         ///< aux time stamp
+
+  // Auxilliary data fields
+  float timestamp; // System time [s]
```

Comment:
would be nice to make all those comments doxygen comments like the ones aboce...